webapp/src/components/expanded_view/call_settings.tsx

```diff
@@ -0,0 +1,477 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+import CCIcon from 'src/components/icons/cc_icon';
+import SettingsWheelIcon from 'src/components/icons/settings_wheel';
+import ShowMoreIcon from 'src/components/icons/show_more';
+import SpeakerIcon from 'src/components/icons/speaker_icon';
+import TickIcon from 'src/components/icons/tick';
+import UnmutedIcon from 'src/components/icons/unmuted_icon';
+import {areLiveCaptionsAvailableInCurrentCall} from 'src/selectors';
+import type {
+    AudioDevices,
+} from 'src/types/types';
+import {getCallsClient} from 'src/utils';
+import styled from 'styled-components';
+
+import ControlsButton from './controls_button';
+
+type AudioDevicesListProps = {
+    deviceType: string;
+    devices: MediaDeviceInfo[];
+    currentDevice: MediaDeviceInfo | null;
+    onDeviceClick: (device: MediaDeviceInfo) => void;
+}
+
+const AudioDevicesList = ({deviceType, devices, currentDevice, onDeviceClick}: AudioDevicesListProps) => {
+    const {formatMessage} = useIntl();
+
+    // Note: this is system default, not the concept of default that we save in local storage in client.ts
+    const makeDeviceLabel = (device: MediaDeviceInfo) => {
+        if (device.deviceId.startsWith('default') && !device.label.startsWith('Default')) {
+            return formatMessage({defaultMessage: 'Default - {deviceLabel}'}, {deviceLabel: device.label});
+        }
+        return device.label;
+    };
```

Comment:
I was going to say: This is one of those places where we should start using useCallback (it can't be put outside the component because we need formatMessage, but we don't need it to be recreated every render).
But then you could say "why not put useCallback around the other functions in this file?" And you could... but there's no real need to -- it's not a hot path at all, and it would probably add even more garbage. So nevermind. Just thinking out loud.

Reply:
Good thoughts. To be honest, I mostly rely on you to tell me if something is highly unoptimized on the React side. But in general, I'd expect us to measure what the actual impact is before making any change and in this instance, it seems minimal considering the whole component is unmounted most of the time.